### PR TITLE
megerd FEATURE_END_USER_REPORTS & FEATURE_ENTITLEMENT into one query

### PR DESCRIPTION
**Pull request by NadavHarnik**
PLAT-5704 #time 3d

### CODE REVIEW

File: alpha/lib/model/PermissionPeer.php [hunk 1]
```diff
@@ -156,6 +156,12 @@ public static function isValidForPartner($permissionName, $partnerId, $checkDepe
 		}
 			
 		$permission = self::getByNameAndPartner($permissionName, array($partnerId, PartnerPeer::GLOBAL_PARTNER));
+		return self::validatePermission($permissionName, $partnerId, $checkDependency, $permission);
+	}
+
+
+	public static function validatePermission($permissionName ,$partnerId, $checkDependency, $permission)
```
Comment (NadavHarnik) [inline alpha/lib/model/PermissionPeer.php]: change to private

File: alpha/lib/model/PermissionPeer.php [hunk 2]
```diff
@@ -209,7 +216,41 @@ public static function getByNameAndPartner($permissionName, $partnerIdsArray)
 		PermissionPeer::setUseCriteriaFilter(true);
 		return $permission;
 	}
-	
+
+	public static function getByNamesAndPartner($permissionNamesArray, $partnerIdsArray)
```
Comment (NadavHarnik) [inline alpha/lib/model/PermissionPeer.php]: enforce array type

File: alpha/lib/model/PermissionPeer.php [hunk 3]
```diff
@@ -209,7 +216,41 @@ public static function getByNameAndPartner($permissionName, $partnerIdsArray)
...
+	{
+		$c = new Criteria();
+
+		if(!is_array($partnerIdsArray))
+			$partnerIdsArray = array($partnerIdsArray);
+
+		if(!in_array('*', $partnerIdsArray, true))
```
Comment (NadavHarnik) [inline alpha/lib/model/PermissionPeer.php]: remove if

File: alpha/lib/model/PermissionPeer.php [hunk 4]
```diff
@@ -209,7 +216,41 @@ public static function getByNameAndPartner($permissionName, $partnerIdsArray)
...
+		{
+			$partnerIdsArray = array_map('strval', $partnerIdsArray);
+			$c->addAnd(PermissionPeer::PARTNER_ID, $partnerIdsArray, Criteria::IN);
+		}
+		if(!in_array('*', $permissionNamesArray, true))
```
Comment (NadavHarnik) [inline alpha/lib/model/PermissionPeer.php]: remove if

File: alpha/lib/model/PermissionPeer.php [hunk 5]
```diff
@@ -209,6 +216,34 @@ public static function getByNameAndPartner($permissionName, $partnerIdsArray)
 		PermissionPeer::setUseCriteriaFilter(true);
 		return $permission;
 	}
+
+	public static function getByNamesAndPartner(array $permissionNamesArray, array $partnerIdsArray)
+	{
+		$c = new Criteria();
+
+		$partnerIdsArray = array_map('strval', $partnerIdsArray);
```
Comment (MosheMaorKaltura) [inline alpha/lib/model/PermissionPeer.php]: This line is redundant - please remove

**Pull request closed without merging**